fdbserver/DataDistribution.actor.cpp

```diff
@@ -667,21 +667,49 @@ struct DDTeamCollection : ReferenceCounted<DDTeamCollection> {
 	}
 
 	~DDTeamCollection() {
-		// The following kills a reference cycle between the teamTracker actor and the TCTeamInfo that both holds and is held by the actor
-		// It also ensures that the trackers are done fiddling with healthyTeamCount before we free this
+		TraceEvent("DDTeamCollectionDestructed", distributorId).detail("Primary", primary);
+		// Other teamCollections also hold pointer to this teamCollection;
+		// TeamTracker may access the destructed DDTeamCollection if we do not reset the pointer
+		for (int i = 0; i < teamCollections.size(); i++) {
+			if (teamCollections[i] != nullptr && teamCollections[i] != this) {
+				for (int j = 0; j < teamCollections[i]->teamCollections.size(); ++j) {
+					if (teamCollections[i]->teamCollections[j] == this) {
+						teamCollections[i]->teamCollections[j] = nullptr;
+					}
+				}
+			}
+		}
+		// Team tracker has pointers to DDTeamCollections both in primary and remote.
+		// The following kills a reference cycle between the teamTracker actor and the TCTeamInfo that both holds and is
+		// held by the actor It also ensures that the trackers are done fiddling with healthyTeamCount before we free
+		// this
 		for(int i=0; i < teams.size(); i++) {
 			teams[i]->tracker.cancel();
 		}
+		// The commented TraceEvent log is useful in detecting what is running during the destruction
+		// TraceEvent("DDTeamCollectionDestructed", distributorId)
```

Comment:
I left these trace events commented instead of removed on purpose. It is useful in figuring out the bug.